dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java

```diff
@@ -121,6 +121,13 @@ protected Result doInvoke(Invocation invocation) throws Throwable {
                     throw getRpcException(type, url, invocation, e);
                 }
             }
+
+            @Override
+            public void destroy() {
+                super.destroy();
+                target.destroy();
+                invokers.remove(this);
+            }

```

Comment:
I think it's unnecessary, because the AbstractProtocol's destroy method has the logic to remove invoker of invokers.